### PR TITLE
Support npu kernel scatter op

### PR DESCRIPTION
**Pull request by oyxuan-11**
### PR types
New features

### PR changes
OPs

### Describe
【NPU】Support npu op scatter

测试通过截图
image

调用NPU截图
image

### CODE REVIEW

File: paddle/fluid/operators/scatter_op_npu.cc [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+           rank, ndims));
+   if (rank < ndims) {
+     std::vector<int64_t> new_dim(ndims, 1);
+     for (int i = ndims - rank; i < ndims; i++) {
+       new_dim[i] = shape[i - ndims + rank];
+     }
+     res.Resize(framework::make_ddim(new_dim));
+   }
+   return res;
+}
```
Comment (zhiqiu) [inline paddle/fluid/operators/scatter_op_npu.cc]: Duplicated with UnsqueezeTo in kron.h? Better just include it.

File: python/paddle/fluid/tests/unittests/npu/test_scatter_op_npu.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                 "core is not compiled with NPU")
+class TestCast1(OpTest):
+    def setUp(self):
+        self.set_npu()
+        self.op_type = "scatter"
+        self.place = paddle.NPUPlace(0)
+
+        #ref_np = np.ones((3, 50)).astype("float32")
+        #index_np = np.array([1, 2]).astype("int32")
+        #updates_np = np.random.random((2, 50)).astype("float32")
```
Comment (zhiqiu) [inline python/paddle/fluid/tests/unittests/npu/test_scatter_op_npu.py]: Remove unused code.

File: python/paddle/fluid/tests/unittests/npu/test_scatter_op_npu.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import paddle.fluid as fluid
+import paddle.fluid.core as core
+
+paddle.enable_static()
+SEED = 2021
+
+
+@unittest.skipIf(not paddle.is_compiled_with_npu(),
+                 "core is not compiled with NPU")
+class TestCast1(OpTest):
```
Comment (zhiqiu) [inline python/paddle/fluid/tests/unittests/npu/test_scatter_op_npu.py:32]: Add more test cases. different dtype, shape, and overwrite or not.

File: paddle/fluid/operators/scatter_op_npu.cc (hunk 1 above)
Comment (oyxuan-11) [reply paddle/fluid/operators/scatter_op_npu.cc]: fixed

File: python/paddle/fluid/tests/unittests/npu/test_scatter_op_npu.py (hunk 2 above)
Comment (oyxuan-11) [reply python/paddle/fluid/tests/unittests/npu/test_scatter_op_npu.py]: fixed

File: python/paddle/fluid/tests/unittests/npu/test_scatter_op_npu.py (hunk 3 above)
Comment (oyxuan-11) [reply python/paddle/fluid/tests/unittests/npu/test_scatter_op_npu.py:32]: fixed

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**